### PR TITLE
memarray: Add functions to extend and reduce the pool with new memory areas

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

This PR adds two functions two manage memarray pools. The `memarray_extend` allows for adding new pool areas to an existing memarray pool. the `memarray_reduce` function can remove (partial) areas from the pool.

I've also added an `memarray_available()` to aid tests.

### Testing procedure

The memarray test application has been extended.

### Issues/PRs references

None

### CODE REVIEW

**Comment by bergzand** [comment]
Rebased to resolve merge conflicts. I've also appended myself to the author list.

**Comment by fjmolinas** [comment]
There are still some conflicts shown on github @bergzand. What use cases do you have for these new functions?

File: sys/include/memarray.h [hunk 1]
```diff
@@ -110,6 +122,44 @@ static inline void memarray_free(memarray_t *mem, void *ptr)
...
+int memarray_reduce(memarray_t *mem, void *data, size_t num);
+
+/**
+ * @brief Returns the number of blocks available
+ *
+ * @param[in]   mem     memarray pool
+ *
+ * @returns             Number of elements available in the memarray pool
+ */
+size_t memarray_available(memarray_t *mem);
```
Comment (fjmolinas) [inline sys/include/memarray.h:161]: This one is nice on itself, I would split into a different commit.

**Comment by bergzand** [comment]
rebased to trigger the new CI checks

File: sys/include/memarray.h (hunk 1 above)
Comment (bergzand) [reply sys/include/memarray.h:161]: Done!

**Comment by bergzand** [comment]
> What use cases do you have for these new functions?

This enables users to create multiple memarray allocators feeding from a larger memarray. This adds a bit of flexibility in that the user doesn't always have to allocate max sized blocks anymore, but can use these smaller block sized allocators. It could even be used to create a rough malloc implementation by having a set of power of 2 block allocators and always allocating a block of the smallest power of 2 bytes still able to fit the requested size.

**Review by fjmolinas — APPROVED** [review]
ACK, looks good and test passes locally, lets wait for murdock!

**Comment by bergzand** [comment]
Squashed in the additional boards that now have insufficient memory
